From 717988daf50b66950dae56a0426e014e044f009f Mon Sep 17 00:00:00 2001
From: Jocelyn Falempe <jocelyn.falempe@intel.com>
Date: Thu, 11 Apr 2013 17:11:27 +0200
Subject: [PORT FROM MAIN] Use new Android.mk rules to build kernel modules

BZ: 99432

Change-Id: Ia05ef8a157bc47bd68434f500359ba253b00da08
Orig-Change-Id: I9a905af8558eb4cb92b37d90a7232688f6bb04bd
Signed-off-by: Jocelyn Falempe <jocelyn.falempe@intel.com>
Reviewed-on: http://android.intel.com:8080/101295
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: Imberton, Guilhem <guilhem.imberton@intel.com>
Tested-by: Imberton, Guilhem <guilhem.imberton@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 wl12xx-compat/AndroidWl12xxCompat.mk |   17 +++++++++--------
 1 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/wl12xx-compat/AndroidWl12xxCompat.mk b/wl12xx-compat/AndroidWl12xxCompat.mk
index 6e70fc9..3f11e8c 100644
--- a/wl12xx-compat/AndroidWl12xxCompat.mk
+++ b/wl12xx-compat/AndroidWl12xxCompat.mk
@@ -1,11 +1,12 @@
 # This makefile is included from vendor/intel/common/wifi/WifiRules.mk.
-
 .PHONY: build_wl12xx-compat
-build_wl12xx-compat: build_kernel
-	+TARGET_TOOLS_PREFIX="$(ANDROID_BUILD_TOP)/$(TARGET_TOOLS_PREFIX)" \
-	TARGET_DEVICE="$(TARGET_DEVICE)" \
-	TARGET_BOARD_PLATFORM="$(TARGET_BOARD_PLATFORM)" \
-	vendor/intel/support/kernel-build.sh \
-	-X hardware/ti/wlan/wl12xx-compat
+TI_SRC_DIR := $(call my-dir)
+TI_BUILD_DIR := $(PRODUCT_OUT)/wl12xx-compat
+
+build_wl12xx-compat: build_bzImage
+	@echo Building kernel module $@
+	$(KERNEL_BLD_ENV) $(MAKE) -C $(TI_SRC_DIR) ARCH=$(KERNEL_ARCH) $(KERNEL_EXTRA_FLAGS) KLIB=$(ANDROID_BUILD_TOP)/$(TI_BUILD_DIR) KLIB_BUILD=$(ANDROID_BUILD_TOP)/$(KERNEL_OUT_DIR) install-modules
+
+copy_modules_to_root: build_wl12xx-compat
 
-$(PRODUCT_OUT)/ramdisk.img : build_wl12xx-compat
+ALL_KERNEL_MODULES += $(TI_BUILD_DIR)
-- 
1.7.4.1

